src/data/mv_polynomial/basic.lean

```diff
@@ -218,6 +218,9 @@ lemma smul_eq_C_mul (p : mv_polynomial σ R) (a : R) : a • p = C a * p := C_mu
 lemma C_eq_smul_one : (C a : mv_polynomial σ R) = a • 1 :=
 by rw [← C_mul', mul_one]
 
+lemma smul_monomial {S₁ : Type*} [monoid S₁] [distrib_mul_action S₁ R] (r : S₁) :
```

Comment:
Note that after #18852 this can be generalized; it probably makes sense to wait.

Reply:
To generalize this to
```lean
lemma smul_monomial {S₁ : Type*} [smul_zero_class S₁ R] (r : S₁) :
  r • monomial s a = monomial s (r • a) := finsupp.smul_single _ _ _
```
maybe an instance should be added:
```lean
instance [comm_semiring R] [smul_zero_class S₁ R] : smul_zero_class S₁ (mv_polynomial σ R) :=
finsupp.smul_zero_class
```

Reply:
That instance looks very reasonable to me; though it should probably be provided for `add_monoid_algebra` too

Reply:
> That instance looks very reasonable to me; though it should probably be provided for `add_monoid_algebra` too

Should I add the instance in this PR, or wait for another PR to add it? (It may be out of range of this PR.)

Reply:
The change has been committed. When I was doing that, I found that `add_monoid_algebra.smul_zero_class` had been added, so I used it for `mv_polynomial.smul_zero_class`.